### ISSUE TITLE
Too general figure styling in .elementor-widget breaks shortcode inclusion of image plugins

### ISSUE THREAD

**Issue by whatsnewsisyphus** [elementor/elementor#7331]
```
.elementor .elementor-widget:not(.elementor-widget-text-editor) figure {
margin: 0 ;
}
```
breaks various image and portfolio plugins as it is a pretty specific selector, overriding their image margins.  You cannot double up on the not arguments so you'll have to come up with a solution otherwise or just drop this declaration altogether, what was it fixing ?

For example, visual portfolio uses

```
.vp-portfolio__item-wrap .vp-portfolio__item {
    margin-left: 40px;
    margin-top: 40px;
}
```
to put a margin between masonry elements. If you add their gallery to elementor using the shortcode box, the images come out with no spacing due to your css blanket.

Somebody tried to report this months ago but their bug-report was super vague and so it got closed.

**Comment by amyschan** [comment]
I'm seeing this as well in a blog post running classic editor.

![2019-03-14_12-20-54 (1)](https://user-images.githubusercontent.com/900909/54385508-0533f880-4654-11e9-8f54-64e9e7890afc.gif)

**Comment by whatsnewsisyphus** [comment]
Thanks for sharing that, yeah unfortunately it will show up in many spots where there are figure elements. I think Astra needs to bite the bullet and overhaul it's CSS system with universal spacing and less blanketing and some semantic structure.

They can cover all the changes in a blog post for all users and push it out, a lot of things seems like they were added organically with features and wordpress changes without an overarching system. The more they wait the more difficult it will be with increasing number of users and features.

**Comment by qwik3r** [comment]
Hope this can be resolved soon, it's a real pain in the ass with Gutenberg which wraps images in figure elements now.

**Comment by jmf3** [comment]
I have to agree with qwik3r. It's shame, elementor is good tool but now unusable for me.

**Comment by netpasja** [comment]
For wordpress gallery margins this works great:

`.single-post .gallery-icon {
margin: 5px;
}`

**Comment by fliteskates** [comment]
Hey everyone,

When I use the image box widget for Elementor it keeps having the same issues as discussed in this thread (text too close to the image).

Is there anything I can do to fix this problem so it will work with the image box widget?

Thx

**Comment by black-eye** [comment]
+1 Also facing these issues

**Comment by Fr4nkHorr1gan** [comment]
> ```
> .elementor .elementor-widget:not(.elementor-widget-text-editor) figure {
> margin: 0 ;
> }
> ```
> breaks various image and portfolio plugins as it is a pretty specific selector, overriding their image margins. You cannot double up on the not arguments so you'll have to come up with a solution otherwise or just drop this declaration altogether, what was it fixing ?
> 
> For example, visual portfolio uses
> 
> ```
> .vp-portfolio__item-wrap .vp-portfolio__item {
>     margin-left: 40px;
>     margin-top: 40px;
> }
> ```
> to put a margin between masonry elements. If you add their gallery to elementor using the shortcode box, the images come out with no spacing due to your css blanket.
> 
> Somebody tried to report this months ago but their bug-report was super vague and so it got closed.

Hello guys,

Made a small fix for Visual Portfolio, very hard but works 
`const replaceCss = () => {`
 `let portfolioItems = $('.vp-portfolio__item');`
 `$(portfolioItems).each(function () {`
 `$(this).replaceWith($('<div class="vp-portfolio__item">' + this.innerHTML + '</div>'));`
` });`
`};`

You can also use some custom class instead of portfolio-item.

**Comment by reelism** [comment]
How has this not been fixed!?

**Comment by joshoastler** [comment]
I'm having this issue too. I have my single post template made in elementor, but want to use Gutenberg editor for the blog post content. It is overwriting the margins on the images for alignment, so I can't center images Ive put in gutenberg editer.
Is there a way to disable this code?
.elementor .elementor-widget:not(.elementor-widget-text-editor) figure

**Comment by Nicoduweb** [comment]
With this bug Elementor is not fully compatible with Gutenberg ! 

To fix this bug with Gutenberg and certainly other plugin i suggest that Elementor Developer replace
`.elementor .elementor-widget:not(.elementor-widget-text-editor) figure`
with
`.elementor .elementor-widget:not(.elementor-widget-text-editor, .elementor-widget-theme-post-content) figure`

This seems to do the job

**Comment by qwik3r** [comment]
I still can't believe Elementor hasn't gotten their ish together and squashed this very SIMPLE BUG. Instead lets have their developers work on creating useless things like absolute positioning elements.. 🙄

**Comment by qwik3r** [comment]
> With this bug Elementor is not fully compatible with Gutenberg !
> 
> To fix this bug with Gutenberg and certainly other plugin i suggest that Elementor Developer replace
> `.elementor .elementor-widget:not(.elementor-widget-text-editor) figure`
> with
> `.elementor .elementor-widget:not(.elementor-widget-text-editor, .elementor-widget-theme-post-content) figure`
> 
> This seems to do the job

You realize you didn't really include the full css code right?

**Comment by Nicoduweb** [comment]
The problem come from the selector not the full CSS.

**Comment by nodakjones** [comment]
This needs to get fixed.  Such a pain with gutenberg

**Comment by glanglois** [comment]
Hi Team,

I am experiencing similar issues as the ones described where I edit my post content on Gutenberg (including some images, centered) and when displayed, they are aligned right.

Is there any timeline on when this could be fixed?

Thnak you. Regards,

**Comment by helvetica123** [comment]
why wrote only "need fixing" "I cant believe this". need to write fuck you all elementor developers . they want this

**Comment by helvetica123** [comment]
> > ```
> > .elementor .elementor-widget:not(.elementor-widget-text-editor) figure {
> > margin: 0 ;
> > }
> > ```
> > 
> > 
> > breaks various image and portfolio plugins as it is a pretty specific selector, overriding their image margins. You cannot double up on the not arguments so you'll have to come up with a solution otherwise or just drop this declaration altogether, what was it fixing ?
> > For example, visual portfolio uses
> > ```
> > .vp-portfolio__item-wrap .vp-portfolio__item {
> >     margin-left: 40px;
> >     margin-top: 40px;
> > }
> > ```
> > 
> > 
> > to put a margin between masonry elements. If you add their gallery to elementor using the shortcode box, the images come out with no spacing due to your css blanket.
> > Somebody tried to report this months ago but their bug-report was super vague and so it got closed.
> 
> Hello guys,
> 
> Made a small fix for Visual Portfolio, very hard but works
> `const replaceCss = () => {`
> `let portfolioItems = $('.vp-portfolio__item');`
> `$(portfolioItems).each(function () {`
> `$(this).replaceWith($('<div class="vp-portfolio__item">' + this.innerHTML + '</div>'));`
> ` });`
> `};`
> 
> You can also use some custom class instead of portfolio-item.

do you write js function when you get every css broken? I cant image your themes.Also probably your customers cant notice the bugs you.

**Comment by shilo-ey** [comment]
This issue has been resolved in Elementor v3.0

Feel free to update

Thanks!
